pkg/providers/ec2.go

```diff
@@ -22,14 +22,14 @@ type EC2Provider struct {
 }
 
 func NewEC2Provider(ctx context.Context, appConfig types.AppConfig, providerRepo repository.ProviderRepository, workerRepo repository.WorkerRepository, tailscale *network.Tailscale) (*EC2Provider, error) {
-	cfg, err := common.GetAWSConfig(appConfig.Providers.EC2Config.AWSAccessKey, appConfig.Providers.EC2Config.AWSSecretKey, appConfig.Providers.EC2Config.AWSRegion, "")
+	cfg, err := common.GetAWSConfig(appConfig.Providers.EC2.AWSAccessKey, appConfig.Providers.EC2.AWSSecretKey, appConfig.Providers.EC2.AWSRegion, "")
```

Comment:
When comparing this provider to Oracle and the others, would this provider be better named as `AWSProvider` instead of `EC2Provider` and the provider config `config.Providers.AWS.xxx` over `config.Providers.EC2.xxx`? Providers are AWS, OCI, GCP whereas their services are EC2 (Elastic Compute Cloud), Compute, and Compute Engine.

Reply:
yea I've been thinking of that as well. I'll rename it in a follow-up PR today